src/gpu/gl/GrGLUtil.cpp

```diff
@@ -586,6 +586,18 @@ GrGLRenderer GrGLGetRenderer(const GrGLInterface* gl) {
     return GrGLGetRendererFromStrings((const char*)rendererString, gl->fExtensions);
 }
 
+bool GrGLGetWebGLRendererSupport(const GrGLInterface* gl) {
+    if (!gl->hasExtension("WEBGL_debug_renderer_info")) {
+        // If we can't get the renderer, then play it safe, because privacy mode could be on.
+        return false;
+    }
```

Comment:
This is the key question: does the extension call here ever fail when we have hardware GPU support?

If we have supported browsers that don't support this, then this check will use the slower path needlessly. 

It might not be a problem -- the slower path might be fast enough, but we don't know that yet, I'm guessing.

Reply:
The slower path still seems decent (it just requires creating an enormous buffer of 0's). And the extension seems to be widely supported - it could be disabled if privacy mode is enabled to block fingerprinting, but in that case I'm playing it safe.

Reply:
okay